src/services/db/TokenService.ts

```diff
@@ -0,0 +1,366 @@
+import { AxiosResponse } from "axios"
+import {
+  Result,
+  ok,
+  err,
+  ResultAsync,
+  okAsync,
+  errAsync,
+  fromPromise,
+} from "neverthrow"
+import { ModelStatic } from "sequelize"
+
+import logger from "@logger/logger"
+
+import DatabaseError from "@errors/DatabaseError"
+import NoAvailableTokenError from "@errors/NoAvailableTokenError"
+import TokenParsingError from "@errors/TokenParsingError"
+
+import { AccessToken } from "@database/models"
+
+export const GITHUB_TOKEN_LIMIT = 5000
+export const GITHUB_TOKEN_THRESHOLD = 4000 // allowed uses
+export const GITHUB_RESET_INTERVAL = 60 * 60 // seconds
+export const GITHUB_TOKEN_REMAINING_HEADER = "x-ratelimit-remaining"
+export const GITHUB_TOKEN_RESET_HEADER = "x-ratelimit-reset"
+const ACTIVE_TOKEN_WARN_LEVEL = 0.6
+const ACTIVE_TOKEN_ALARM_LEVEL = 0.8
+const GITHUB_TOKEN_LENGTH = 40
+
+export type MaybeResetTime = Result<number, null>
+export const NoResetTime: MaybeResetTime = err(null)
+
+export type TokenData = {
```

Comment:
don't have to do it here, but 1 thing i noticed was that there's a clear semantic difference between `ReservedTokens` and `ActiveTokens`, which wasn't really clear. 

we can instead expose a `transitionToReserved` function that takes in `ActiveToken` and returns a `ReservedToken` and does some db stuff under the hood too

Reply:
Introduced semantic differences between `ActiveTokenData` and `ReservedTokenData`.